src/tools/compiletest/src/lib.rs

```diff
@@ -718,7 +720,7 @@ fn make_test(
     inputs: &Stamp,
     poisoned: &mut bool,
 ) -> Vec<test::TestDescAndFn> {
-    let test_path = if config.mode == Mode::RunMake {
+    let test_path = if config.mode == Mode::RunMake && !testpaths.file.join("rmake.rs").exists() {
```

Comment:
I think you will also need a case for extracting these comments from `rmake.rs`.